compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/LangLibrary.java

```diff
@@ -0,0 +1,225 @@
+
+    private final Map<String, Map<String, BInvokableSymbol>> langLibMethods;
+    private final Map<String, List<FunctionSymbol>> wrappedLangLibMethods;
+    private final SymbolFactory symbolFactory;
+
+    private LangLibrary(CompilerContext context) {
+        context.put(LANG_LIB_KEY, this);
+
+        symbolFactory = SymbolFactory.getInstance(context);
+        SymbolTable symbolTable = SymbolTable.getInstance(context);
+        Map<String, BPackageSymbol> langLibs = new HashMap<>();
+        wrappedLangLibMethods = new HashMap<>();
+
+        for (Map.Entry<BPackageSymbol, SymbolEnv> entry : symbolTable.pkgEnvMap.entrySet()) {
+            BPackageSymbol module = entry.getKey();
+            PackageID moduleID = module.pkgID;
+
+            if (Names.BALLERINA_ORG.equals(moduleID.orgName) &&
+                    (moduleID.nameComps.size() == 2 && Names.LANG.equals(moduleID.nameComps.get(0)))) {
+                langLibs.put(moduleID.nameComps.get(1).value, module);
+            }
+        }
+
+        langLibMethods = getLangLibMethods(langLibs);
+    }
+
+    public static LangLibrary getInstance(CompilerContext context) {
+        LangLibrary langLib = context.get(LANG_LIB_KEY);
+        if (langLib == null) {
+            langLib = new LangLibrary(context);
+        }
+
+        return langLib;
+    }
+
+    /**
+     * Given a type descriptor kind, return the list of lang library functions that can be called using a method call
+     * expr, on an expression of that type.
+     *
+     * @param typeDescKind A type descriptor kind
+     * @return The associated list of lang library functions
+     */
+    public List<FunctionSymbol> getMethods(TypeDescKind typeDescKind) {
+        String langLibName = getAssociatedLangLibName(typeDescKind);
+
+        if (wrappedLangLibMethods.containsKey(langLibName)) {
+            return wrappedLangLibMethods.get(langLibName);
+        }
+
+        Map<String, BInvokableSymbol> methods = langLibMethods.get(langLibName);
+
+        List<FunctionSymbol> wrappedMethods = new ArrayList<>();
+        wrappedLangLibMethods.put(langLibName, wrappedMethods);
+        populateMethodList(wrappedMethods, methods);
+
+        // Add the common functions in lang.value to types which have an associated lang library.
+        if (!LANG_VALUE.equals(langLibName)) {
+            populateMethodList(wrappedMethods, langLibMethods.get(LANG_VALUE));
+        }
+
+        return wrappedMethods;
+    }
+
+    // Private Methods
+
+    private void populateMethodList(List<FunctionSymbol> list, Map<String, BInvokableSymbol> langLib) {
+        for (Map.Entry<String, BInvokableSymbol> entry : langLib.entrySet()) {
+            FunctionSymbol method = symbolFactory.createFunctionSymbol(entry.getValue(), entry.getKey());
+            list.add(method);
+        }
+    }
+
+    private String getAssociatedLangLibName(TypeDescKind typeDescKind) {
+        switch (typeDescKind) {
+            case INT:
+            case BYTE:
```

Comment:
We can simplify the cases and the returns here. What if get the common cases to the bottom where we can directly call getName() over the typeDescKind